tests/core/stream.c

```diff
@@ -17,13 +17,29 @@ static int test_ctor(git_stream **out, const char *host, const char *port)
 	return 0;
 }
 
+static int test_stream_wrap(git_stream **out, git_stream *in, const char *host)
+{
+	GIT_UNUSED(in);
+	GIT_UNUSED(host);
+
+	ctor_called = 1;
+	*out = &test_stream;
+
+	return 0;
+}
+
 void test_core_stream__register_tls(void)
 {
 	git_stream *stream;
+	git_stream_registration registration = {0};
 	int error;
 
+	registration.version = 42;
```

Comment:
Honestly, stream registration should fail if we see an unsupported version as it indicates that the application invoking the registration may have an incompatible structure definition

Reply:
Agreed.